nav2_controller/src/nav2_controller.cpp

```diff
@@ -51,27 +51,36 @@ DwbController::DwbController()
     }, costmap_ros_);
 }
 
-DwbController::~DwbController()
+nav2_controller::~nav2_controller()
 {
   RCLCPP_INFO(get_logger(), "Destroying");
   costmap_executor_.cancel();
   costmap_thread_->join();
 }
 
 nav2_util::CallbackReturn
-DwbController::on_configure(const rclcpp_lifecycle::State & state)
+nav2_controller::on_configure(const rclcpp_lifecycle::State & state)
 {
-  RCLCPP_INFO(get_logger(), "Configuring");
+  std::string local_planner_name;
+
+  RCLCPP_INFO(get_logger(), "Configuring local planner interface");
 
   costmap_ros_->on_configure(state);
 
   auto node = shared_from_this();
 
   progress_checker_ = std::make_unique<ProgressChecker>(rclcpp_node_);
 
-  planner_ = std::make_unique<dwb_core::DWBLocalPlanner>(
-    node, costmap_ros_->getTfBuffer(), costmap_ros_);
-  planner_->on_configure(state);
+  try {
+    // TODO(yathartha3): Use Parameter server correctly
+    local_planner_ = lp_loader_.createUniqueInstance("dwb_core::DWBLocalPlanner");
+    RCLCPP_INFO(get_logger(), "Created local_planner");
+    local_planner_->initialize(node, costmap_ros_);
+    local_planner_->on_configure(state);
+  } catch (const pluginlib::PluginlibException & ex) {
```

Comment:
We had talked about loading the goal checker here, and then using it in this package instead of in dwb. Were you leaving that to a follow on PR?

Reply:
@crdelsey Yes, I will make those changes in a follow up PR.